src/adagrams.js

```diff
@@ -1,15 +1,190 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
+const letterCount = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
 };
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+const letterValue = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
 };
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+class Adagrams {
+  
+  static letterPool() {
+    // using the object.keys to get the array string of letters
+    let key = Object.keys(letterCount);
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 1
-};
+    // create the array of letters
+    let letterPool = [];
+
+    key.forEach((letter) => {
+      for (let i = 0; i < letterCount[letter]; i++) {
+        // getting the values how many times string letter appear in the array
+        // EX: X: 2 -> ['X', 'X]
+        letterPool.push(letter);
+      }
+    });
+    return letterPool;
+  };
+
+  static drawLetters() {
+    // create an empty array
+    let handLetters = [];
+    // create a copy and using 3 dot and new set method in array so we don't get the duplicate letters
+    let letterPoolCopy = [...new Set(Adagrams.letterPool())];
```

Comment:
I like the idea here, but I think we actually do want to be able to get duplicates! For example, our `letterCount` says we can have up to 12 `E`s.